### PR TITLE
Update Loss Functions to Accept Tensor num_items_in_batch

### PR DESCRIPTION
**Pull request by iamsernine**
This PR updates the ForCausalLMLoss and ForMaskedLMLoss functions to handle num_items_in_batch as a torch.Tensor instead of an int, aligning with how the parameter is actually passed during training.

Changes:

Updated num_items_in_batch: Optional[int] → Optional[torch.Tensor] in both functions.

Applied .item() conversion when computing loss to retain compatibility with fixed_cross_entropy.

### CODE REVIEW

**Comment by Rocketknight1** [comment]
cc @sunmarc I think!

File: src/transformers/loss/loss_utils.py [hunk 1]
```diff
@@ -61,6 +61,10 @@ def ForCausalLMLoss(
     shift_labels = shift_labels.view(-1)
     # Enable model parallelism
     shift_labels = shift_labels.to(logits.device)
+
+    if num_items_in_batch is not None:
+        num_items_in_batch = num_items_in_batch.item()
```
Comment (SunMarc) [inline src/transformers/loss/loss_utils.py:66]: this will trigger a device sync (making training slower) so let's just move the tensor to the right device instead

File: src/transformers/loss/loss_utils.py [hunk 2]
```diff
@@ -78,9 +82,12 @@ def ForMaskedLMLoss(
 
     # Flatten the tokens
     logits = logits.view(-1, vocab_size)
-    labels = labels.view(-1)
+    labels = labels.view(-1).to(logits.device)
     # Enable model parallelism
 
+    if num_items_in_batch is not None:
+        num_items_in_batch = num_items_in_batch.item()
+
```
Comment (SunMarc) [inline src/transformers/loss/loss_utils.py:90]: same here

File: src/transformers/loss/loss_utils.py [hunk 3]
```diff
@@ -78,9 +82,12 @@ def ForMaskedLMLoss(
 
     # Flatten the tokens
     logits = logits.view(-1, vocab_size)
-    labels = labels.view(-1)
+    labels = labels.view(-1).to(logits.device)
```
Comment (SunMarc) [inline src/transformers/loss/loss_utils.py:85]: no need for that as we do it below

File: src/transformers/loss/loss_utils.py [hunk 4]
```diff
@@ -78,9 +82,12 @@ def ForMaskedLMLoss(
...
     # Enable model parallelism
 
+    if num_items_in_batch is not None:
+        num_items_in_batch = num_items_in_batch.item()
```
Comment (SunMarc) [inline src/transformers/loss/loss_utils.py:89]: can you also update the docstring above ?

**Review by SunMarc — DISMISSED** [review]
Thanks ! Left a couple of comments

**Comment by NEREUScode** [comment]
@SunMarc  you mean i should pull the first changes i made before ?

**Comment by SunMarc** [comment]
> you mean i should pull the first changes i made before ?

The changes you made before were a bit better, try to follow the indication I gave you

**Comment by NEREUScode** [comment]
> > you mean i should pull the first changes i made before ?
> 
> The changes you made before were a bit better, try to follow the indication I gave you

i'll finish some work and i'll open a new PR

**Comment by SunMarc** [comment]
Just update this PR instead of opening a new one !

**Review by iamsernine — COMMENTED** [review]
i forget how i change this but it's the same as this

**Review by ArthurZucker — APPROVED** [review]
thanks

**Comment by iamsernine** [comment]
@ArthurZucker thanks feel free to tag me if there anything i can help with
